components/chart/canvas/index.tsx

```diff
@@ -130,10 +131,20 @@ function Canvas({
   useEffect(() => {
     toggleScrolling(isScrolling);
 
+    if (searchedNode && canvasRef.current) {
+      const { clientHeight, clientWidth } = canvasRef.current;
+      const newScrollTop = Math.max(0, searchedNode.y - clientHeight / 2);
+      const newScrollLeft = Math.max(0, searchedNode.x - clientWidth / 2);
+      canvasRef.current.scrollTop = newScrollTop;
+      canvasRef.current.scrollLeft = newScrollLeft;
+    }
```

Comment:
It's always better to have a separate useEffect for each functionality while their dependencies are not correlated